multipaz/src/commonMain/kotlin/org/multipaz/mdoc/mso/MobileSecurityObjectParser.kt

```diff
@@ -216,6 +218,9 @@ class MobileSecurityObjectParser(
             require(validUntil > validFrom) {
                 "The validUntil timestamp should be later than the validFrom timestamp"
             }
+            require(currentTime <= validUntil) {
+                "The MSO has expired and cannot be cryptographically authenticated. validUntil: $validUntil, current time: $currentTime"
+            }
```

Comment:
It's not appropriate to do this check here, `MobileSecurityObjectParser` is just a mechanism to parse MSOs and convey to the user of the API what's in it. It's entirely possible that this API is used on the wallet side where you could have expired MSOs. Or you could use it on the server-side where the clock could be wrong.

I get what you're saying that checks like these need to be done but they need to be done in a verifier app, not the low-level routine for parsing an MSO.